package/libsolv.spec.in

```diff
@@ -15,122 +15,97 @@
+BuildRequires:  pkgconfig(zlib)
+%if %{with ruby}
+BuildRequires:  ruby
+BuildRequires:  ruby-devel
+BuildRequires:  swig
 %endif
 %if 0%{?fedora} || 0%{?rhel} >= 6 || 0%{?mageia}
 BuildRequires:  db-devel
 %endif
-BuildRequires:  libxml2-devel
-%if 0%{?suse_version} && 0%{?suse_version} < 1100
-BuildRequires:  graphviz
-%endif
-%if 0%{?suse_version} > 1020 || 0%{?fedora} || 0%{?mageia}
-BuildRequires:  fdupes
-%endif
-BuildRequires:  cmake
-BuildRequires:  gcc-c++
-BuildRequires:  rpm-devel
-BuildRequires:  zlib-devel
-
-%if %{with perl_binding}
+%if %{with perl}
 BuildRequires:  perl
+BuildRequires:  swig
 %if 0%{?fedora} || 0%{?rhel} >= 6 || 0%{?mageia}
 BuildRequires:  perl-devel
 %endif
-BuildRequires:  swig
 %endif
-
-%if %{with ruby_binding}
-%global ruby_vendorarch %(ruby  -r rbconfig -e "puts RbConfig::CONFIG['vendorarchdir'].nil? ? RbConfig::CONFIG['sitearchdir'] : RbConfig::CONFIG['vendorarchdir']")
-BuildRequires:  ruby
-BuildRequires:  ruby-devel
+%if %{with python2}
 BuildRequires:  swig
+BuildRequires:  pkgconfig(python2)
 %endif
-
-%if %{with python_binding}
-%global python_sitearch %(python -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(True))")
-BuildRequires:  python-devel
+%if %{with python3}
 BuildRequires:  swig
+BuildRequires:  pkgconfig(python3)
 %endif
-
-%if %{with python3_binding}
-%global python3_sitearch %(python3 -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(True))")
-BuildRequires:  python3-devel
-%endif
-
 %if %{with bz2}
-%if 0%{?suse_version}
-BuildRequires:  libbz2-devel
-%else
-BuildRequires:  bzip2-devel
-%endif
+BuildRequires:  pkgconfig(bzip2)
 %endif
-
 %if %{with xz}
-BuildRequires:  xz-devel
+BuildRequires:  pkgconfig(liblzma)
 %endif
 
-
-Summary:        Package dependency solver using a satisfiability algorithm
-License:        BSD-3-Clause
-Group:          Development/Libraries/C and C++
-
 %description
 libsolv is a library for solving packages and reading repositories.
 The solver uses a satisfiability algorithm.
 
 %if !%{with disable_shared}
-%package -n libsolv@LIBSOLV_SOVERSION@
+%package -n %{libname}
 Summary:        Package dependency solver using a satisfiability algorithm
 Group:          System/Libraries
 
-%description -n libsolv@LIBSOLV_SOVERSION@
+%description -n %{libname}
 libsolv is a library for solving packages and reading repositories.
 It consists of two central blocks: Using a dictionary approach to
 store and retrieve package and dependency information, and, using a
 so-called satisfiability algorithm for resolving package
 dependencies.
-
 %endif
+
 %package devel
 Summary:        Development files for libsolv, a package solver
 Group:          Development/Libraries/C and C++
+Requires:       pkgconfig(rpm)
+Conflicts:      libsatsolver-devel
 %if !%{with disable_shared}
-Requires:       libsolv@LIBSOLV_SOVERSION@ = %version
+Requires:       %{libname} = %{version}
 %endif
-Requires:       rpm-devel
```

Comment:
Could we also add a `Requires: %{name}-tools = %{version}-%{release}` so that we can get rid of the awkward `%exclude` in the tools subpackage file list for SUSE on helix2solv to randomly move it to devel?